test/samplers/delayed_sampling.js

```diff
@@ -0,0 +1,223 @@
+      const stateKey = 'PrioritySampler'; // TODO ideally should be uniqueName() per BaseSamplerB2
+      let state: ?PrioritySamplerState = store[stateKey];
+      if (!state) {
+        state = {
+          samplerFired: Array(this._delegates.length),
+        };
+        // cannot use array.fill() in Node 0.10
+        for (let i = 0; i < this._delegates.length; i++) {
+          state.samplerFired[i] = false;
+        }
+        store[stateKey] = state;
+      }
+      return state;
+    }
+
+    onCreateSpan(span: Span): SamplingDecision {
+      const state = this._getState(span);
+      let retryable = false;
+      for (let i = 0; i < this._delegates.length; i++) {
+        if (state.samplerFired[i]) {
+          continue;
+        }
+        const d = this._delegates[i].onCreateSpan(span);
+        retryable = retryable || d.retryable;
+        if (!d.retryable) {
+          state.samplerFired[i] = true;
+        }
+        if (d.sample) {
+          return d; // TODO do we want to alter out tags?
+        }
+      }
+      return { sample: false, retryable: retryable, tags: null };
+    }
+
+    onSetOperationName(span: Span, operationName: string): SamplingDecision {
+      // FIXME:
+      return this.onCreateSpan(span);
+    }
+
+    onSetTag(span: Span, key: string, value: any): SamplingDecision {
+      // FIXME:
+      return this.onCreateSpan(span);
+    }
+
+    toString(): string {
+      return 'DelegatingSampler';
+    }
+
+    close(callback: ?() => void): void {
+      const countdownCallback = Utils.countdownCallback(this._delegates.length, callback);
+      this._delegates.forEach(r => r.close(countdownCallback));
+    }
+  }
+
+  describe('with PrioritySampler', () => {
+    const tagSampler = new TagEqualsSampler('theWho', [{ tagValue: 'Bender' }, { tagValue: 'Leela' }]);
+    const constSampler = new ConstSampler(false);
+    const priSampler = new PrioritySampler([tagSampler, constSampler]);
+    const reporter = new InMemoryReporter();
+    const tracer = new Tracer('test-service-name', reporter, priSampler);
+
+    beforeEach(() => {});
+
+    it('should not sample or finalize new span without tags', () => {
+      let span = tracer.startSpan('opName');
+      assert.isFalse(span._spanContext.isSampled(), 'sampled');
+      assert.isFalse(span._spanContext.samplingFinalized, 'finalized');
+    });
+
+    it('should sample and finalize created span with tag', () => {
+      let span = tracer.startSpan('opName', { tags: { theWho: 'Bender' } });
+      assert.isTrue(span._spanContext.isSampled(), 'sampled');
+      assert.isTrue(span._spanContext.samplingFinalized, 'finalized');
+    });
+
+    it('should sample and finalize span after setTag', () => {
+      let span = tracer.startSpan('opName');
+      assert.isFalse(span._spanContext.isSampled(), 'sampled');
+      assert.isFalse(span._spanContext.samplingFinalized, 'finalized');
+      span.setTag('theWho', 'Leela');
```

Comment:
Is this wip? Seems to be missing asserts

Reply:
fixed